### PR TITLE
feat(store,world): use user-types for `ResourceId`, `FieldLayout` and `Schema` in table libraries

### PR DESCRIPTION
**Pull request by alvrs**
Thanks to #1566 we can now use the real user-types in table libraries and don't need to manually `wrap`/`unwrap` everywhere

### CODE REVIEW

File: packages/store/gas-report.json [hunk 1]
```diff
@@ -603,73 +603,73 @@
     "file": "test/StoreCoreDynamic.t.sol",
     "test": "testGetDynamicFieldSlice",
     "name": "get field slice (cold, 1 slot)",
-    "gasUsed": 10240
+    "gasUsed": 5609
```
Comment (alvrs) [inline packages/store/gas-report.json:606]: these gas changes come from not loading the `_tableId` and `_keyTuple` from storage in the gas test

Comment (holic) [reply packages/store/gas-report.json:606]: same for all tests or just this one? I noticed some other gas improvements below

**Review by holic — DISMISSED** [review]
So nice!

File: packages/store-sync/src/common.ts [hunk 2]
```diff
@@ -76,7 +76,12 @@ export type StorageAdapterBlock = { blockNumber: BlockLogs["blockNumber"]; logs:
 export type StorageAdapter = (block: StorageAdapterBlock) => Promise<void>;
 
 // TODO: adjust when we get namespace support (https://github.com/latticexyz/mud/issues/994) and when table has namespace key (https://github.com/latticexyz/mud/issues/1201)
-export const schemasTable = storeConfig.tables.Tables;
+// TODO: adjust when schemas are automatically resolved
+export const schemasTable = {
+  ...storeConfig.tables.Tables,
+  valueSchema: resolveUserTypes(storeConfig.tables.Tables.valueSchema, storeConfig.userTypes),
+};
+
```
Comment (alvrs) [inline packages/store-sync/src/common.ts:84]: kind of annoying, but we need this workaround for now to land the contract changes (until we have a better way to resolve the user type config with strong types)

File: packages/store/test/StoreCoreDynamic.t.sol [hunk 3]
```diff
@@ -96,41 +98,43 @@ contract StoreCoreDynamicTest is Test, GasReporter, StoreMock {
...
-    assertEq(StoreCore.getField(_tableId, _keyTuple, 2, fieldLayout), thirdDataBytes);
+    assertEq(bytes32(StoreCore.getField(tableId, keyTuple, 0, fieldLayout)), firstDataBytes);
+    assertEq(StoreCore.getField(tableId, keyTuple, 2, fieldLayout), thirdDataBytes);
   }
 
   function testPopFromThirdField() public {
     FieldLayout fieldLayout = StoreCore.getFieldLayout(_tableId);
     bytes memory dataBytes = thirdDataBytes;
+    ResourceId tableId = _tableId;
+    bytes32[] memory keyTuple = _keyTuple;
```
Comment (dk1a) [inline packages/store/test/StoreCoreDynamic.t.sol:137]: we could use immutable too, it's about as convenient as storage, but without the gas overhead. It's only useful for constant-like values tho

**Review by dk1a — DISMISSED** [review]

File: packages/store-sync/src/common.ts (hunk 2 above)
Comment (holic) [reply packages/store-sync/src/common.ts:84]: hopefully coming in #1561

**Review by holic — DISMISSED** [review]

File: packages/store/gas-report.json (hunk 1 above)
Comment (alvrs) [reply packages/store/gas-report.json:606]: `wrapping`/`unwrapping` in itself should not significantly impact gas, so i think most of the gas variation in here comes from the changed gas tests

**Pull request merged**